browser/resources/settings/getting_started_page/brave_account_common.ts

```diff
@@ -0,0 +1,23 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+export function onEyeIconClicked(event: Event) {
+  event.preventDefault()
+  let type = 'password'
+  const target = event.target as Element
+  target.setAttribute('name',
+    target.getAttribute('name')!.replace(/[^-]*$/, match => {
+      if (match === 'off') {
+        type = 'text'
+        return 'on'
+      }
+      return 'off'
+    })
+  )
+  target.parentElement!.setAttribute('type', type)
+}
+
+const emailRegexp = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/
```

Comment:
bold :laughing:


```suggestion
export const emailRegexp = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/
```

Reply:
If we're not re-using the browser's own built-in validator, then all we should do in terms of validation is:
- ensuring that the email address contains at least one `@` sign and rejecting it otherwise (and at least one character before & after)
- rejecting any email with non-ASCII characters

The actual validation will take place when the user clicks the link. See https://docs.google.com/document/d/1luuOB2aZtrJjVrKLbOiC0EVowLa8olvqmDBTaYm35lw/edit?tab=t.0 for the reasoning if interested.

Reply:
Agreed and we have a similar check in components/common as far as I remember

Reply:
We do have something similar:

https://github.com/brave/brave-core/blob/9684ea1164558b6f02d1695e5995cb8241cada61/components/common/isValidEmailAddress.ts#L12-L28

But it's not checking:
- whether there's at least 1 character before & after the `@` sign
- for non-ASCII characters

Do we want to add these checks to this function and use it here?

Reply:
> But it's not checking:
> 
> - whether there's at least 1 character before & after the @ sign
> - for non-ASCII characters
> 
> Do we want to add these checks to this function and use it here?

That sounds good to me.

We should also update the above function to check that the number of `@` signs is `>= 1`. The comment is wrong and a valid email could have more than one `@` sign.

Reply:
@fmarier Would this check be sufficient?
```js
export function isEmailAddressValid(input: string) {
  if (!input) {
    // input is falsy
    return false
  }

  if (!/^[\x00-\x7F]+$/.test(input)) {
    // input contains non-ASCII
    return false
  }

  const index = input.lastIndexOf('@')
  if (index === -1) {
    // there's no @ sign in input
    return false
  }

  if (input.substring(0, index) === "" || input.substring(index + 1) === "") {
    // there are no characters either before or after the last @ sign
    return false
  }

  return true
}
```

Reply:
Looks good to me.